bootloader/src/pyi_main.c

```diff
@@ -129,408 +95,434 @@ pyi_main(PYI_CONTEXT *pyi_ctx)
      * to read run-time options */
 
-    /* For the curious:
-     * On Windows, the UTF-8 form of MEIPASS2 is passed to pyi_setenv, which
-     * decodes to UTF-16 before passing it to the Windows API. So the var's value
-     * is full unicode.
-     *
-     * On OS X/Linux, the MEIPASS2 value is passed as the bytes received from the OS.
-     * Only Python will care about its encoding, and it is passed to Python using
-     * PyUnicode_DecodeFSDefault.
-     */
-
-    extractionpath = pyi_getenv("_MEIPASS2");
-
-    /* NOTE: record the in-child status here, because extractionpath
-     * might get overwritten later on (on Windows and macOS, single
-     * process is used for --onedir mode). */
-    in_child = (extractionpath != NULL);
-    if (in_child) {
-        /* Check if _PYI_ONEDIR_MODE is set to 1; this is set by linux/unix
-         * bootloaders when they restart themselves within the same process
-         * to achieve single-process onedir execution mode. This case should
-         * be treated as if extractionpath was not set at this point yet,
-         * i.e., in_child needs to be reset to 0. */
-        char *pyi_onedir_mode = pyi_getenv("_PYI_ONEDIR_MODE");
-        if (pyi_onedir_mode) {
-            if (strcmp(pyi_onedir_mode, "1") == 0) {
-                in_child = 0;
-            }
-            free(pyi_onedir_mode);
-            pyi_unsetenv("_PYI_ONEDIR_MODE");
-        }
-    }
-
-    /* If the Python program we are about to run invokes another PyInstaller
-     * one-file program as subprocess, this subprocess must not be fooled into
-     * thinking that it is already unpacked. Therefore, PyInstaller deletes
-     * the _MEIPASS2 variable from the environment.
-     */
+    pyi_ctx->is_onefile = pyi_ctx->archive->needs_to_extract;
+    pyi_ctx->needs_to_extract = pyi_ctx->is_onefile;
 
-    pyi_unsetenv("_MEIPASS2");
+    /* These are passed on to python interpreter, so they show up in sys.argv */
+    pyi_ctx->archive->argc = pyi_ctx->argc;
+    pyi_ctx->archive->argv = pyi_ctx->argv;
 
-    VS("LOADER: _MEIPASS2 is %s\n", (extractionpath ? extractionpath : "not set"));
+    /* Read argv emulation setting (macOS .app bundles) */
+#if defined(__APPLE__) && defined(WINDOWED)
+    pyi_ctx->macos_argv_emulation = pyi_arch_get_option(pyi_ctx->archive, "pyi-macos-argv-emulation") != NULL;
+#endif
 
+    /* Read and decode console hiding/minimization option (Windows only) */
 #if defined(_WIN32) && !defined(WINDOWED)
-    /* Early console hiding/minimization */
-    const char *hide_console_option = pyi_arch_get_option(pyi_ctx->archive_status, "pyi-hide-console");
-    if (hide_console_option != NULL) {
-        if (strcmp(hide_console_option, HIDE_CONSOLE_OPTION_HIDE_EARLY) == 0) {
-            pyi_win32_hide_console();
-        } else if (strcmp(hide_console_option, HIDE_CONSOLE_OPTION_MINIMIZE_EARLY) == 0) {
-            pyi_win32_minimize_console();
+    if (1) {
```

Comment:
Ah, yes... I tend to use `if (1) { }` for additional scoping instead of `{ }`.  We can do a search for all `if (1)` and remove them, if you find them too offending.

Reply:
Meh, they can stay. I just thought they might be forgotten placeholders for genuine conditionals.